### PR TITLE
Add libgcc for linux

### LINKED ISSUE #16: libgcc missing run requirement on linux?
I think `libgcc` is a missing run requirement on linux. 

I tested this on a debian docker container.

On a new miniconda installation I get this:
```
>>> import spacy
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/conda/lib/python3.6/site-packages/spacy/__init__.py", line 5, in <module>
    from .deprecated import resolve_model_name
  File "/opt/conda/lib/python3.6/site-packages/spacy/deprecated.py", line 8, in <module>
    from .cli import download
  File "/opt/conda/lib/python3.6/site-packages/spacy/cli/__init__.py", line 5, in <module>
    from .train import train, train_config
  File "/opt/conda/lib/python3.6/site-packages/spacy/cli/train.py", line 8, in <module>
    from ..scorer import Scorer
  File "/opt/conda/lib/python3.6/site-packages/spacy/scorer.py", line 4, in <module>
    from .gold import tags_to_entities
ImportError: libgomp.so.1: cannot open shared object file: No such file or directory
```

If I install libgcc from defaults it works:

```
root@26bbe6af725a:/work# conda install libgcc
Fetching package metadata ...........
Solving package specifications: .

Package plan for installation in environment /opt/conda:

The following NEW packages will be INSTALLED:

    libgcc: 5.2.0-0

Proceed ([y]/n)? y

libgcc-5.2.0-0 100% |##################################################################################################################################| Time: 0:00:00   4.39 MB/s
root@26bbe6af725a:/work# python
Python 3.6.0 |Continuum Analytics, Inc.| (default, Dec 23 2016, 12:22:00)
[GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import spacy
>>>
```

I can submit a PR to fix this but I wanted to know if there is something I am missing.

### PR DESCRIPTION
**Pull request by danielfrg**
Fixes #16

### CODE REVIEW

**Comment by danielfrg** [comment]
@rmax could you take a quick look at this?

**Comment by jakirkham** [comment]
So normally we don't use `libgcc` unless `gcc` is used to build it. As far as `gcc` we only use it if we need Fortran or OpenMP. The latter can also be gotten with `clangdev`. Are either of those two things needed here? If not, we don't need `libgcc`. If so, we need `gcc` and not `libgcc` in `build`.

**Comment by rmax** [comment]
> @jakirkham: Are either of those two things needed here?

As reported by @danielfrg, `spacy`'s compiled modules depend on `libgomp`. This also is reported by `conda inspect`:

```
spacy
-----

python-3.6.1-2:
    libpython3.6m.so.1.0 (lib/libpython3.6m.so.1.0)

system:
    libc.so.6 (/lib/x86_64-linux-gnu/libc.so.6)
    libdl.so.2 (/lib/x86_64-linux-gnu/libdl.so.2)
    libgcc_s.so.1 (/lib/x86_64-linux-gnu/libgcc_s.so.1)
    libgomp.so.1 (/usr/lib/x86_64-linux-gnu/libgomp.so.1)
    libm.so.6 (/lib/x86_64-linux-gnu/libm.so.6)
    libpthread.so.0 (/lib/x86_64-linux-gnu/libpthread.so.0)
    librt.so.1 (/lib/x86_64-linux-gnu/librt.so.1)
    libstdc++.so.6 (/usr/lib/x86_64-linux-gnu/libstdc++.so.6)
    libutil.so.1 (/lib/x86_64-linux-gnu/libutil.so.1)
    linux-vdso.so.1 ()

not found:
```

So, if the system has not the gcc libraries it the module will fail to load. Below is the output of `conda inspect` after installing `libgcc`:

```
spacy
-----

libgcc-5.2.0-0:
    libgcc_s.so.1 (lib/libgcc_s.so.1)
    libgomp.so.1 (lib/libgomp.so.1)
    libstdc++.so.6 (lib/libstdc++.so.6)

python-3.6.1-2:
    libpython3.6m.so.1.0 (lib/libpython3.6m.so.1.0)

system:
    libc.so.6 (/lib/x86_64-linux-gnu/libc.so.6)
    libdl.so.2 (/lib/x86_64-linux-gnu/libdl.so.2)
    libm.so.6 (/lib/x86_64-linux-gnu/libm.so.6)
    libpthread.so.0 (/lib/x86_64-linux-gnu/libpthread.so.0)
    librt.so.1 (/lib/x86_64-linux-gnu/librt.so.1)
    libutil.so.1 (/lib/x86_64-linux-gnu/libutil.so.1)
    linux-vdso.so.1 ()

not found:
```

Also I found this issue reported on spacy tracker https://github.com/explosion/spaCy/issues/1110

To provide a smooth user experience, I think we need to add `gcc` as build requirement and `libgcc` as runtime requirement for `linux`.

@ines @honnibal thoughts?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -27,9 +27,10 @@ requirements:
     - setuptools
     - numpy x.x
     - numpy >=1.7
-
+    - libgcc     # [linux]
```
Comment (rmax) [inline recipe/meta.yaml]: @danielfrg please change this to `gcc`.

Comment (danielfrg) [inline recipe/meta.yaml]: I have changed `libgcc` to `gcc` for build.

**Comment by jakirkham** [comment]
Thanks for the info @rmax. Sounds like the right approach.

**Comment by danielfrg** [comment]
I have changed libgcc to gcc for build.

**Pull request merged**